### PR TITLE
DOC: Update the docs of InterpolatedUnivariateSpline and LSQUnivariateSpline. <fitpack2>.

### LINKED ISSUE #14098: DOC-Update for InterpolatedUnivariateSpline and LSQUnivariateSpline
My issue is about inconsistency in the nomenclature of the  inherited classes:

1. `InterpolatedUnivariateSpline(UnivariateSpline):`
2. `LSQUnivariateSpline(UnivariateSpline):`

The inconsistency has two sources:

1. Default is not consistent defined as shown in the issue #14093
2. Default values are not defined like in case of parameter `k`:
https://github.com/scipy/scipy/blob/31351394351f944b9cf87c1fbc54213192cfe180/scipy/interpolate/fitpack2.py#L593-L594
`k` should be 3.

#### Reproducing code example:

https://github.com/scipy/scipy/blob/31351394351f944b9cf87c1fbc54213192cfe180/scipy/interpolate/fitpack2.py#L573-L611

https://github.com/scipy/scipy/blob/31351394351f944b9cf87c1fbc54213192cfe180/scipy/interpolate/fitpack2.py#L687-L730

#### Scipy/Numpy/Python version information:

Current master branch

### PR DESCRIPTION
**Pull request by Anselmoo**
#### Reference issue

Closes #14098.

#### What does this implement/fix?

1 .  It removed double white space in class `UnivariateSpline`.
2.  It updates the parameter section in the docs of `LSQUnivariateSpline(UnivariateSpline)`, `InterpolatedUnivariateSpline(UnivariateSpline)`.
3. Correct the spacing for `bbox`.

### CODE REVIEW

File: scipy/interpolate/fitpack2.py
Comment (tupui) [inline scipy/interpolate/fitpack2.py:90]: Previous case is correct as when you pass a parameter to a function you don't have spaces between the equal sign. Same bellow.

The case with ``k = 3`` vs ``k=3`` is more arguable I think as this could read more as *math*.

**Review by tupui — CHANGES_REQUESTED** [review]
These kind of change are very cosmetic. We don't have strong guidelines on how we should write these things (talking about: Default is ... vs The default value is ... vs If ... (default)). What is required is to have the definition of what is the default value.

I would still consider this PR as it brings a bit of uniformity (and you are a new contributor, which we are always grateful 😃), **but** if you had planned to make more PR like these I would ask you to group the changes into this PR.

We tend to discourage such PR: it takes time to review and clutter git blame for little added value. See the note paragraph of this section of our contributing guide https://scipy.github.io/devdocs/dev/hacking.html#maintaining-existing-code

File: scipy/interpolate/fitpack2.py
Comment (Anselmoo) [reply scipy/interpolate/fitpack2.py:90]: I agree; according to your suggestion; line 100
```
 If `s` is None, ``s = len(w)`` which should be a good value if
```
has to be change, too??

Comment (tupui) [reply scipy/interpolate/fitpack2.py:90]: Yes. And now that I think about this more, please change back all `k = 3` to `k=3` (and alike). Let's just be consistent with the fact that this is a parameter and how it's called.

**Review by tupui — APPROVED** [review]
LGTM, thanks @Anselmoo. Do you have other changes planned before I merge?

**Comment by Anselmoo** [comment]
> LGTM, thanks @Anselmoo. Do you have other changes planned before I merge?

@tupui thx for the support! In terms of the class `UnivariateSpline`.

In general, there are still some inconsistencies like see below, because default values are not defined. This should maybe become better a new issue like DOC-check of module <fitpack2>??

https://github.com/scipy/scipy/blob/a6b0a01484f428f83790995b78487b73d706c198/scipy/interpolate/fitpack2.py#L326-L341

**Comment by tupui** [comment]
> In general, there are still some inconsistencies like see below, because default values are not defined. This should maybe become better a new issue like DOC-check of module <fitpack2>??

As I said, do not create other issues for this, as it creates unnecessary work for all maintainers. You can address them here as it's the same thing. A PR can touch multiple file/folders.

**Comment by Anselmoo** [comment]
> > In general, there are still some inconsistencies like see below, because default values are not defined. This should maybe become better a new issue like DOC-check of module ??
> 
> As I said, do not create other issues for this, as it creates unnecessary work for all maintainers. You can address them here as it's the same thing. A PR can touch multiple file/folders.

Ic, that differs from some other communities. Then I will go ahead and fix it. I don't saw it in the first place!!

File: scipy/interpolate/fitpack2.py
Comment (tupui) [inline scipy/interpolate/fitpack2.py:2022]: No it needs a float as per the doc. This is passing an int instead.

File: scipy/interpolate/fitpack2.py
Comment (tupui) [inline scipy/interpolate/fitpack2.py:1882]: This information is put after along our doc and not capitalized.

File: scipy/interpolate/fitpack2.py
Comment (tupui) [inline scipy/interpolate/fitpack2.py:381]: It should be without the space actually.

File: scipy/interpolate/fitpack2.py
Comment (tupui) [inline scipy/interpolate/fitpack2.py:438]: Avoid unnecessary changes like this one. Here it would not change anything so please revert such changes.

File: scipy/interpolate/fitpack2.py
Comment (tupui) [inline scipy/interpolate/fitpack2.py:694]:
```suggestion
    Fits a spline ``y=spl(x)`` of degree `k` to the provided `x`, `y` data. `t`
```

File: scipy/interpolate/fitpack2.py
Comment (tupui) [inline scipy/interpolate/fitpack2.py:1044]: The description bellow is more accurate. Just missing a point but it is accurate.

File: scipy/interpolate/fitpack2.py
Comment (tupui) [inline scipy/interpolate/fitpack2.py:1141]: I am not sure this would render properly, or might be missing indentation.

File: scipy/interpolate/fitpack2.py
Comment (tupui) [inline scipy/interpolate/fitpack2.py:1326]: Having both parameter under the same description was valid and preferred.

File: scipy/interpolate/fitpack2.py
Comment (tupui) [inline scipy/interpolate/fitpack2.py:1339]: Same

File: scipy/interpolate/fitpack2.py
Comment (tupui) [inline scipy/interpolate/fitpack2.py:1509]: Same

**Review by tupui — CHANGES_REQUESTED** [review]
This is a lot of changes, thanks. While some changes might be good (missing parameters), I am not sure about a few things.

@ev-br I would prefer to have your input on this one.

File: scipy/interpolate/fitpack2.py
Comment (Anselmoo) [reply scipy/interpolate/fitpack2.py:438]: I agree with the part `x`. In case of `der` vs. `d`, the function `derivatives` returns just d and not `der`. I personally find it a little missleading

File: scipy/interpolate/fitpack2.py
Comment (Anselmoo) [reply scipy/interpolate/fitpack2.py:381]: I agree

File: scipy/interpolate/fitpack2.py
Comment (Anselmoo) [reply scipy/interpolate/fitpack2.py:1141]: You are right!

File: scipy/interpolate/fitpack2.py
Comment (Anselmoo) [reply scipy/interpolate/fitpack2.py:1044]: Shouldn't be it `dxi` and not `dx`?

File: scipy/interpolate/fitpack2.py
Comment (Anselmoo) [reply scipy/interpolate/fitpack2.py:1860]: Sorry I don't get it

Comment (tupui) [reply scipy/interpolate/fitpack2.py:1860]: Sorry what I mean is: `array_like (n,)`
For the most up to date best practices we have, I suggest you have a look at recently added code such as `stats.bootstrap` or `stats.qmc`.

File: scipy/interpolate/fitpack2.py
Comment (tupui) [reply scipy/interpolate/fitpack2.py]: I did not check the code, but just reading the change would give me the impression that x and y could be 2 points.

File: scipy/interpolate/fitpack2.py
Comment (Anselmoo) [reply scipy/interpolate/fitpack2.py:1860]: Ic [stats.qmc](https://github.com/scipy/scipy/blob/master/scipy/stats/_qmc.py#L527-L678) and I agree but this will cover then all (N, ) in fitpack2. I will take care

Comment (tupui) [reply scipy/interpolate/fitpack2.py]: You changed it to `(n,) array_like` instead of `array_like (n,)`.

File: scipy/interpolate/fitpack2.py
Comment (tupui) [reply scipy/interpolate/fitpack2.py:438]: Still it should be coherent with the code here. As I said, we should not change this as it would not do anything.

**Comment by tupui** [comment]
@rgommers did you intentionally close the PR or is this a side effect of the renaming? I've seen a few PR/issues I am following getting closed at the same time.

**Comment by rgommers** [comment]
See https://github.com/scipy/scipy/pull/9291#issuecomment-1004248815 for the explanation and follow-up.

**Comment by rgommers** [comment]
This is a very weird GitHub bug though, because https://github.com/Anselmoo/scipy still exists!

**Comment by tupui** [comment]
I see thanks. Yes sounds like a bug otherwise all PRs should be gone basically.

**Comment by rgommers** [comment]
Ah, the message when hovering over this branch here is wrong:

image

The repo still exists, but the branch is not present. Please revive the PR if that makes sense (see https://github.com/scipy/scipy/pull/9291#issuecomment-1004248815)

**Comment by tupui** [comment]
Hum if you filter the PRs by `is:pr is:closed sort:updated-desc`. There are quite a few... Well if that can help cleaning stalled PRs...

**Comment by rgommers** [comment]
I checked a few - in some cases the forked repo was gone (e.g., gh-12307), in other cases the branch was gone (this one is odd, because I thought removing a branch would already have auto-closed the PR).

**Comment by tupui** [comment]
Do you think we should send a quick email to warn maintainers and link maybe to the list of recently closed PRs so we all can have a look and quickly assess if PRs can stay closed or open?

**Comment by rgommers** [comment]
Ah, was already typing that email:) Sent before I saw this.

**Comment by jarrodmillman** [comment]
We had this happen on the NX repo, too.  All the PRs that got closed for us were due to the originating PR's repo or branch being deleted.

**Pull request closed without merging**
